File: Largest-Local-Values-in-a-Matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n = grid.size();
    vector<vector<int>> result(n - 2, vector<int>(n - 2));
    
    for (int i = 0; i < n - 2; i++) 
    {
        for (int j = 0; j < n - 2; j++) 
        {
            int maxVal = 0;
            for (int x = i; x < i + 3; x++) 
            {
                for (int y = j; y < j + 3; y++) 
                {
                    if (grid[x][y] > maxVal) 
                    {
                        maxVal = grid[x][y];
                    }
                }
            }
            result[i][j] = maxVal;
        }
    }
    return result;
}
};

File: Largest-Local-Values-in-a-Matrix.py
class Solution(object):
    def largestLocal(self, grid):
        n = len(grid)
        result = []

        for i in range(n - 2):
            row = []
            for j in range(n - 2):
                maxVal = 0
                for x in range(i, i + 3):
                    for y in range(j, j + 3):
                        if grid[x][y] > maxVal:
                            maxVal = grid[x][y]
                row.append(maxVal)
            result.append(row)

        return result
